### PR TITLE
Local terminal reconnection

### PR DESCRIPTION
**Pull request by meganrogge**
This PR is part of #20013

### CODE REVIEW

File: src/vs/workbench/contrib/terminal/browser/terminalService.ts [hunk 1]
```diff
@@ -200,19 +202,67 @@ export class TerminalService implements ITerminalService {
...
 	}
 
-	private attachRemoteListeners(): void {
+	private _reconnectToLocalTerminals(): void {
+		// Reattach to all local terminals
+		const defaultLayoutInfo = JSON.stringify({ tabs: [{ terminals: [], isActive: true }] });
+		const result = this._storageService.get('localTerminalLayoutInfo', StorageScope.WORKSPACE, defaultLayoutInfo);
+		const layoutInfo = JSON.parse(result);
+
+		if (layoutInfo.tabs.length === 0) {
```
Comment (rheh) [inline src/vs/workbench/contrib/terminal/browser/terminalService.ts:215]: We are checking 'if (layoutInfo) {' on line 222 but accessing its properties before this check occurs.  Is this right?

File: src/vs/workbench/contrib/terminal/browser/terminalService.ts [hunk 2]
```diff
@@ -200,19 +202,67 @@ export class TerminalService implements ITerminalService {
...
+			this.createTerminal();
+			this.attachListeners(false);
+			return;
+		}
+
+		let activeTab: ITerminalTab | undefined;
+		if (layoutInfo) {
+			for (const layout of layoutInfo.tabs) {
+				let terminalInstance: ITerminalInstance | undefined;
```
Comment (rheh) [inline src/vs/workbench/contrib/terminal/browser/terminalService.ts:224]: Would it help to move this section into a new function e.g. this._openTabTerminals(layout)

File: src/vs/workbench/contrib/terminal/browser/terminalService.ts (hunk 1 above)
Comment (meganrogge) [reply src/vs/workbench/contrib/terminal/browser/terminalService.ts:215]: thanks for the review, but this is not close to being merged/is a draft for that reason.

File: src/vs/workbench/contrib/terminal/browser/terminalService.ts [hunk 3]
```diff
@@ -200,19 +202,67 @@ export class TerminalService implements ITerminalService {
...
-		this.attachRemoteListeners();
+		this.attachListeners(true);
 		this._onDidChangeConnectionState.fire();
 	}
 
-	private attachRemoteListeners(): void {
+	private _reconnectToLocalTerminals(): void {
+		// Reattach to all local terminals
+		const defaultLayoutInfo = JSON.stringify({ tabs: [{ terminals: [], isActive: true }] });
+		const result = this._storageService.get('localTerminalLayoutInfo', StorageScope.WORKSPACE, defaultLayoutInfo);
```
Comment (rheh) [inline src/vs/workbench/contrib/terminal/browser/terminalService.ts:212]: Is result a specific type? e.g. IStorageService?

File: src/vs/workbench/contrib/terminal/browser/terminalService.ts (hunk 1 above)
Comment (rheh) [reply src/vs/workbench/contrib/terminal/browser/terminalService.ts:215]: Ooops, missed the draft status.

**Pull request merged**